### PR TITLE
Allow arbitrary data to be passed to templates

### PR DESCRIPTION
**Pull request by MarcDufresne**
This follows a discussion from #1470

### CODE REVIEW

File: Documentation/templates.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Steps:
+
+1. Copy contents of the `web` directory over to a new directory.
+2. Customize the templates as needed, be sure to retain all the existing variables so Dex continues working correctly.
+  a. Use this syntax `{{ "your_key" | extra }}` to use values from `frontend.extra`.
+3. Write a theme for your templates in the `themes` directory.
+4. Add your custom data to the Dex configuration `frontend.extra`.
+   ```yaml
+   frontend:
```
Comment (srenatus) [inline Documentation/templates.md]: Would you mind adding a `frontend` section to the test in `cmd/dex/config_test.go`?

File: Documentation/templates.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+     dir: /path/to/custom/web
+     extra:
+       foo: "bar"
```
Comment (srenatus) [inline Documentation/templates.md]: I know this goes back to me -- but any more realistic example would be cool here. Your usage might be a good one?

**Comment by srenatus** [comment]
Also let's please squash these before merging. :)

**Comment by MarcDufresne** [comment]
Tests added and docs updated! Will you squash merge on GitHub, or you want me to squash first?

**Comment by srenatus** [comment]
Thanks, can you please do the squashing? :)

**Pull request merged**